webdriverio/web/examples/typescript/test/pageobjects/login.page.ts

```diff
@@ -0,0 +1,23 @@
+import Page from './page';
+
+/**
+ * sub page containing specific selectors and methods for a specific page
+ */
+export class LoginPage extends Page {
+    /**
+     * a method to encapsule automation code to interact with the page
+     * e.g. to login using username and password
+     */
+    login (username: string, password: string) {
+        $('#username').setValue(username);
```

Comment:
I would advice you to be consistent with putting your selectors outside of the methods and even make them private like this

Reply:
@wswebcreation what is the advantage of creating a getter if it's not going to allow access to the property?

Reply:
A getting in JS will allow you to call a method like this `this.username` instead of `this.username()` . The getter will still allow you to access it outside of your PO

Reply:
@wswebcreation yes, it's the same concept as other programming languages. Can you please help me to understand the advantage of creating a getter for a locator that shouldn't be accessible outside of the PO?
Furthermore, what is the advantage of using this code from a getter instead of just invoking `$('#username')` directly?

Reply:
This PO is now very small, but if you have multiple methods that also use the same selector you only have one reference. Because you want it to be an example on how to do it I would suggest putting it as a getting. Secondly, personally I'm never a fan of having selectors in my methods. A PO for me should start with the selectors and then methods and the methods should have a ref to the selectors.